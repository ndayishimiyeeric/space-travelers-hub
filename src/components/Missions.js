import React from 'react';
import { useSelector } from 'react-redux';
import Mission from './Mission';
import styles from './styles/Missions.module.css';

const Missions = () => {
  const missions = useSelector((state) => state.mission);
  const missionElements = missions.map((mission) => (
    <Mission
      key={mission.id}
      id={mission.id}
      missionName={mission.mission_name}
      description={mission.description}
      active={mission.active}
    />
  ));

  return (
    <>
      <p className={styles.header}>Mission Name</p>
      <p className={styles.header}>Description</p>
      <p className={styles.header}>Status</p>
      <p className={styles.header}>Active</p>

      { missionElements }
    </>
  );
};

export default Missions;
