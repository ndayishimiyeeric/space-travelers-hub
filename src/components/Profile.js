import React from 'react';
import { useSelector } from 'react-redux';
import styles from './styles/Profile.module.css';

function Profile() {
  const rockets = useSelector((state) => state.rockets);
  const missions = useSelector((state) => state.mission);
  const rocketElement = rockets.map((rocket) => (
    rocket.active === true ? <li className={styles.element}>{rocket.rocket_name}</li> : null));
  const missionElement = missions.map((mission) => (
    mission.active === true ? <li className={styles.element}>{mission.mission_name}</li> : null
  ));
  return (
    <div className={styles.profileDiv}>
      <div className={styles.left}>
        <h4>My Missions</h4>
        <ul className={styles.elements}>
          { missionElement }
        </ul>
      </div>
      <div className={styles.right}>
        <h4>My Rockets</h4>
        <ul className={styles.elements}>
          { rocketElement }
        </ul>
      </div>
    </div>
  );
}

export default Profile;
