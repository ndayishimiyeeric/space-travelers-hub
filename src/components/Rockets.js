import React from 'react';
import { useSelector } from 'react-redux';
import Rocket from './Rocket';
import styles from './styles/Rockets.module.css';

function Rockets() {
  const rockets = useSelector((state) => state.rockets);
  const rocketElements = rockets.map((rocket) => (
    <Rocket
      key={rocket.id}
      id={rocket.id}
      img={rocket.img}
      description={rocket.description}
      title={rocket.rocket_name}
      active={rocket.active}
    />
  ));
  return (
    <div className={styles.rocketsDiv}>
      { rocketElements }
    </div>
  );
}

export default Rockets;
