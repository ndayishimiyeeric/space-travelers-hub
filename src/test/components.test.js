import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../redux/configureStore';
import Missions from '../components/Missions';
import Rockets from '../components/Rockets';
import Profile from '../components/Profile';

describe('Test Compontests', () => {
  test('test Rockets', () => {
    const tree = renderer
      .create(<Provider store={store}><Rockets /></Provider>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('test My Profile Link', () => {
    const tree = renderer
      .create(<Provider store={store}><Missions /></Provider>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('test My Profile', () => {
    const tree = renderer
      .create(<Provider store={store}><Profile /></Provider>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
