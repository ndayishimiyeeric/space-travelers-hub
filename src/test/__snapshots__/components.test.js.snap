// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Compontests test My Profile 1`] = `
<div
  className="profileDiv"
>
  <div
    className="left"
  >
    <h4>
      My Missions
    </h4>
    <ul
      className="elements"
    />
  </div>
  <div
    className="right"
  >
    <h4>
      My Rockets
    </h4>
    <ul
      className="elements"
    />
  </div>
</div>
`;

exports[`Test Compontests test My Profile Link 1`] = `
<div
  className="missionDiv"
>
  <div
    className="heading"
  >
    <p>
      Mission Name:
    </p>
    <p>
      Description:
    </p>
    <p>
      Status:
    </p>
    <p>
      Reserve Mission:
    </p>
  </div>
</div>
`;

exports[`Test Compontests test Rockets 1`] = `
<div
  className="rocketsDiv"
/>
`;
